.github/actions/arm64-integration/action.yml

```diff
@@ -0,0 +1,34 @@
+name: ARM64 integration tests
+description: Runs the deep integration tests on self-hosted ARM64 runner
+
+inputs:
+  tag:
+    required: true
+  docker-registry:
+    required: true
+  k3d-version:
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f
+      with:
+        go-version: '1.18'
+    - uses: docker/setup-buildx-action@94ab11c41e45d028884a99163086648e898eed25
+    - shell: bash
+      run: |
+        bin/docker-pull-binaries "${{ inputs.tag }}"
+        CMD="$PWD/target/release/linkerd2-cli-${{ inputs.tag}}-linux-arm64"
+        echo "CMD=$CMD" >> "$GITHUB_ENV"
+        "$CMD" version --client
+    - uses: azure/setup-kubectl@7f7e5ba5ea3e491b92e6e8e5819963f8f3a1f076
+    - shell: bash
+      env:
+        TAG: ${{ inputs.tag }}
+        RUN_ARM_TEST: 1
+        LINKERD_DOCKER_REGISTRY: ${{ inputs.docker-registry }}
+      run: bin/tests --name deep --images preload "$CMD"
+    - shell: bash
+      run: bin/k3d cluster delete --all
+      if: always()
```

Comment:
Why is this necessary? Does the worker not automatically clean up resources created by the action?

Reply:
Nope, this runs directly into the server and no automatic cleanup is performed.
`bin/tests` by default cleans up the cluster but not when the suite fails.
Tested in 0ce4351

Reply:
To further clarify, there's no container/vm setup here that would make things transient.

Reply:
Do we want to consider keeping the cluster around when tests fail? Flakiness could be annoying, but if they are failing consistently it could be nice to be able to SSH into the runner for further debugging.

Reply:
@kleimkuhler What do you think about having the cleanup happen before the tests get triggered? If the tests succeed they'll get cleaned up by default anyway, if they fail one would be able to see what happened and the cluster would get cleaned before the next run.
Note though that a lot of tests are wrapped around `TestHelper.WithDataPlaneNamespace` which kills the namespace when the tests finishes (successfully or not). I think it'd be a nice improvement (outside this PR) to bypass the killing when the test fails.

Reply:
So we cleanup old clusters before running a new set of tests? That sounds good to me. If things fail then we'll have time to debug further if we need to.

And yep it's unfortunate we cleanup Namespaces with that function even if the tests fail, but I agree it's better to sort that out separately!

Reply:
Eh, I'd prefer that we cleanup by default. If you're in the situation where you want to rerun it you can always modify the workflow to disable the cleanup.

Can we randomize the cluster name so that we have no possibility of reusing the same cluster? Or would that require test changes?

Reply:
> Can we randomize the cluster name so that we have no possibility of reusing the same cluster? Or would that require test changes?

The cluster receives the name of its test, in this case `deep`. We'd have to change the testing library (`bin/_test-helpers.sh`) to override that, perhaps through a flag.

Reply:
That sounds good to me @mateiidavid . I've pushed d48b21b using just recipies and the required setup. I've also replaced the CLUSTER_NAME snippet with the branch/tag name as discussed off-github. Note that once the PR gets accepted I'll move this in to `release.yaml` to minimize the chance something's wrong.